### PR TITLE
Make Segment Compile On Windows

### PR DESCRIPTION
**Pull request by brianmichel**
Adjust the Segment codebase to be buildable on Windows.

### CODE REVIEW

**Comment by brianmichel** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #7** <a href="https://app.graphite.dev/github/pr/thebrowsercompany/analytics-swift/7?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #8** <a href="https://app.graphite.dev/github/pr/thebrowsercompany/analytics-swift/8?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/thebrowsercompany/analytics-swift/7?utm_source=stack-comment).

**Comment by brianmichel** [comment]
@compnerd when I test this PR in CI and on my local I see the same test failures (yay consistency), but the errors that I see if I turn on additional logging locally seem to be focused around libcurl and ssl, is this possibly just an issue with this toolchain build?

```
Test Case 'FlushPolicyTests.testIntervalBasedFlushPolicy' started at 2023-11-28 14:39:54.022
noneCleaned up 0 non-running uploads.
noneUploads in-progress: 0
noneProcessing Batch:
0-segment-events.temp
noneError uploading request Protocol "https" not supported or disabled in libcurl.
errorAn internal error occurred: networkUnknown(Error Domain=NSURLErrorDomain Code=-1002 "(null)")
noneProcessed: 0-segment-events.temp
noneCleaned up 0 non-running uploads.
noneCleaned up 1 non-running uploads.
noneUploads in-progress: 0
noneProcessing Batch:
0-segment-events.temp
noneError uploading request Protocol "https" not supported or disabled in libcurl.
errorAn internal error occurred: networkUnknown(Error Domain=NSURLErrorDomain Code=-1002 "(null)")
noneProcessed: 0-segment-events.temp
noneCleaned up 0 non-running uploads.
```

**Comment by compnerd** [comment]
libcurl shouldn't be exposed - that is indirectly only accessible through `FoundationNetworking`.  We do disable SSL as we use WinSSL so that the system configuration is made available.

**Comment by compnerd** [comment]
@brianmichel and I paired a bit on this.  This is a regression since 5.9.1.  There is a fix in flight for this.

**Comment by brianmichel** [comment]
> @brianmichel and I paired a bit on this. This is a regression since 5.9.1. There is a fix in flight for this.

Link to the fix in question, https://github.com/apple/swift/pull/70077

**Comment by brianmichel** [comment]
@compnerd looks like the Swift PR merged! Do you happen to know which snapshot it might start appearing in so I can adjust CI on this PR?

**Comment by compnerd** [comment]
@brianmichel since you are using thebrowsercompany/swift-build: 20231129.2

**Comment by brianmichel** [comment]
@compnerd @darinf I believe this is ready for review. There are two tests that seems pretty flaky around storage, but I don't think those should hold us back on merging this personally. 

@compnerd the `testPurgeStorage` uses the purge functions on the SDK to basically just remove files on disk. I know we've seen other issues with file manipulation and foundation, could this just be another bug?

File: Examples/other_plugins/NotificationTracking.swift [hunk 1]
```diff
@@ -33,15 +33,15 @@
 
 // MARK: Common
 
-#if !os(Linux) && !os(macOS)
+#if !os(Linux) && !os(macOS) && !os(Windows)
```
Comment (compnerd) [inline Examples/other_plugins/NotificationTracking.swift:36]: Is this supposed to just be `#if os(iOS)`?

**Review by compnerd — APPROVED** [review]

**Comment by compnerd** [comment]
> @compnerd the testPurgeStorage uses the purge functions on the SDK to basically just remove files on disk. I know we've seen other issues with file manipulation and foundation, could this just be another bug?

Plausible!  I wouldn't count out any bugs in Foundation.  Would be good to understand the exact usage and make sure that it isn't that case.

File: Examples/other_plugins/NotificationTracking.swift (hunk 1 above)
Comment (brianmichel) [reply Examples/other_plugins/NotificationTracking.swift:36]: I believe they also support tvOS and watchOS (I think), maybe visionOS as well? It seemed safer to keep their pattern of just using the negation.

**Comment by brianmichel** [comment]
### Merge activity

* **Dec 4, 5:13 PM**: @brianmichel started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thebrowsercompany/analytics-swift/7).
* **Dec 4, 5:13 PM**: @brianmichel merged this pull request with [Graphite](https://app.graphite.dev/github/pr/thebrowsercompany/analytics-swift/7).

**Pull request merged**